backend/infrahub/core/diff/merger/serializer.py

```diff
@@ -267,19 +267,26 @@ def _get_default_property_merge_dicts(self, action: DiffAction) -> dict[Database
             ),
         }
 
-    def _get_added_removed_peers(
-        self, relationship_diff: EnrichedDiffSingleRelationship
-    ) -> tuple[str | None, str | None]:
+    def _get_actions_and_peers(self, relationship_diff: EnrichedDiffSingleRelationship) -> list[tuple[DiffAction, str]]:
         is_related_prop = [p for p in relationship_diff.properties if p.property_type is DatabaseEdgeType.IS_RELATED][0]
         actions_and_values = self._get_property_actions_and_values(property_diff=is_related_prop, python_value_type=str)
-        added = None
-        removed = None
+        actions_and_peers: list[tuple[DiffAction, str]] = []
         for action, peer_id in actions_and_values:
             if action is DiffAction.ADDED:
-                added = str(peer_id)
+                actions_and_peers.append((DiffAction.ADDED, str(peer_id)))
             elif action is DiffAction.REMOVED:
-                removed = str(peer_id)
-        return added, removed
+                actions_and_peers.append((DiffAction.REMOVED, str(peer_id)))
+
+        conflict = relationship_diff.conflict
+        if (
+            conflict
+            and conflict.selected_branch
+            and conflict.selected_branch is ConflictSelection.DIFF_BRANCH
+            and conflict.base_branch_value
+            and conflict.base_branch_action in (DiffAction.ADDED, DiffAction.UPDATED)
+        ):
```

Comment:
if there is a conflict and the diff version is selected, we need to delete the version on main, which we were not doing before